### ISSUE TITLE
Security Issue with Key Cloak 6.0.1

### ISSUE THREAD

**Issue by GraceW-qac** [QAC-Portal/QA-Portal#113]
When signing in, part or all of the token is printed into the URL which leaves it subject to a man-in-the -middle attack. Part of the digital signature is printed.

An unknown user could authorise themselves and access other user data with the issue above.

This is also an issue regarding GDPR compliance.

**Comment by GraceW-qac** [comment]
- [x] Find if there is a workaround to stop Key Cloak from printing the CRL into the URL.

Find if there is a means of encoding digital signatures so if they are printed, they are printed in their ciphertext(found to be unnecessary as I have stopped KeyCloak from printing the access token).

**Comment by Lothandar** [comment]
https://snyk.io/vuln/SNYK-JAVA-ORGKEYCLOAK-449844
I found some documentation about the issue: apparently 6.0.1 is not affected but it's the one we are running on.
Edit V6.0.1 of https://mvnrepository.com/artifact/org.keycloak/keycloak-server-spi-private/6.0.1
fix the problem I am still unsure if this is installed on the standalone version we are running (the standalone shouldn't be use for deployment anyway so to check if the full build fix the problem)

**Comment by GraceW-qac** [comment]
I can confirm that 6.0.1 is affected by the issue above, in testing and in the patchnotes for this version.

I have found two possible methods to prevent keycloak from printing the access token to the URL.

- [x] Begin implementation of the methods described above.

If they break anything, try different means of implementation or look into changing the token to ciphertext (also found to be unnecessary as the issue has been resolved by preventing KeyCloak from printing the access token).
